staging/src/k8s.io/apimachinery/pkg/watch/filter.go

```diff
@@ -87,10 +87,14 @@ var _ Interface = &Recorder{}
 // NewRecorder wraps an Interface and records any changes sent across it.
 func NewRecorder(w Interface) *Recorder {
 	r := &Recorder{}
-	r.Interface = Filter(w, r.record)
+	r.SetInterface(w)
 	return r
 }
 
+func (r *Recorder) SetInterface(w Interface) {
```

Comment:
this is a little difficult to reason about... what are the thread safety guarantees around this? if Stop() or ResultChan() was invoked on the previous interface, do events from that interface continue to get recorded in the background?

Reply:
No thread safety guarantees, I should have made that clear with a comment. This is used in the case where the watcher fails and the RetryWatcher creates a new one to update watcher for the Recorder and because for the first watcher is actually created after the Recorder.